### DISCUSSION TITLE
run a script in SpaceClaim using run_discovery_script_file

### DISCUSSION THREAD

**Discussion by XIAO-GUO-commint** [ansys/pyansys-geometry#871] [Q&A] [answered]
I want to run a script in SpaceClaim using run_discovery_script_file, but I'm unsure what to include in script_args. I have reviewed the original documentation, but it did not provide guidance on how to manage this.
from ansys.geometry.core import launch_modeler_with_spaceclaim

modeler_discovery = launch_modeler_with_spaceclaim()

modeler_discovery.run_discovery_script_file(file_path=r'E:\2023project\nncx\muti-optimize\Mesh_Gem\1.scscript',script_args=script_args)
![Uploading image.png…]()

**Comment by RobPasMue** [comment, marked as answer]
Hi @XIAO-GUO-commint - this is related to issue #870

As of now, the argument is mandatory - please pass in an empty dictionary (i.e. ``{}``) in case you do not want to provide any arguments to the script. We are working on making this argument optional.
